packages/dynamic_layouts/test/wrap_layout_test.dart

```diff
@@ -0,0 +1,371 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:dynamic_layouts/dynamic_layouts.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+void main() {
+  testWidgets(
+      'DynamicGridView generates children and checks if they are layed out',
+      (WidgetTester tester) async {
+    final List<Widget> children = List<Widget>.generate(
+      10,
+      (int index) => SizedBox(
+        height: index.isEven ? 100 : 50,
+        width: index.isEven ? 95 : 180,
+        child: Center(child: Text('Item $index')),
+      ),
+    );
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Scaffold(
+          body: DynamicGridView(
+            gridDelegate: const SliverGridDelegateWithWrapping(),
+            children: children,
+          ),
+        ),
+      ),
+    );
+    expect(find.text('Item 0'), findsOneWidget);
+    for (int i = 0; i < 10; i++) {
+      expect(find.text('Item $i'), findsOneWidget);
+    }
+  });
+
+  testWidgets(
+      'Test for wrap that generates children and checks if they are layed out',
+      (WidgetTester tester) async {
+    final List<Widget> children = List<Widget>.generate(
+      10,
+      (int index) => SizedBox(
+        height: index.isEven ? 100 : 50,
+        width: index.isEven ? 95 : 180,
+        child: Center(child: Text('Item $index')),
+      ),
+    );
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Scaffold(
+          body: DynamicGridView.wrap(
+            children: children,
+          ),
+        ),
+      ),
+    );
+    for (int i = 0; i < 10; i++) {
+      expect(find.text('Item $i'), findsOneWidget);
+    }
+  });
+
+  testWidgets('Test for wrap to be laying child dynamically',
+      (WidgetTester tester) async {
+    final List<Widget> children = List<Widget>.generate(
+      20,
+      (int index) => SizedBox(
+        height: index.isEven ? 1000 : 50,
+        width: index.isEven ? 95 : 180,
+        child: Center(child: Text('Item $index')),
+      ),
+    );
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Scaffold(
+          body: DynamicGridView.builder(
+              itemCount: children.length,
+              gridDelegate: const SliverGridDelegateWithWrapping(),
+              itemBuilder: (
+                BuildContext context,
+                int index,
+              ) =>
+                  children[index]),
```

Comment:
ad a trailing comma before `),` to get better formatting?

Reply:
(Here and elsewhere in the file)

Reply:
I ended up removing the second comma, I think it looks better
<img width="497" alt="Screen Shot 2022-08-29 at 17 25 05" src="https://user-images.githubusercontent.com/65375294/187309610-7cef3842-32cc-4cb8-a9a6-4166ea095ef5.png">